qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java

```diff
@@ -178,6 +176,122 @@ public void test70BindMountCustomPathConfAndJvmOptions() throws Exception {
+     */
+    public void test80SetEnvironmentVariablesUsingFiles() throws Exception {
+        final Path secretsDir = getTempDir().resolve("secrets");
+        final String optionsFilename = "esJavaOpts.txt";
+
+        try {
+            mkdir(secretsDir);
+
+            // ES_JAVA_OPTS_FILE
+            Files.writeString(secretsDir.resolve(optionsFilename), "-XX:-UseCompressedOops\n");
+
+            Map<String, String> envVars = Map.of("ES_JAVA_OPTS_FILE", "/run/secrets/" + optionsFilename);
+
+            // File permissions need to be secured in order for the ES wrapper to accept
+            // them for populating env var values
+            Files.setPosixFilePermissions(secretsDir.resolve(optionsFilename), p600);
+
+            final Map<Path, Path> volumes = Map.of(secretsDir, Path.of("/run/secrets"));
+
+            // Restart the container
+            runContainer(distribution(), volumes, envVars);
+
+            waitForElasticsearch(installation);
+
+            final String nodesResponse = makeRequest(Request.Get("http://localhost:9200/_nodes"));
+
+            assertThat(nodesResponse, containsString("\"using_compressed_ordinary_object_pointers\":\"false\""));
+        } finally {
+            rm(secretsDir);
+        }
+    }
+
+    /**
+     * Check that the elastic user's password can be configured via a file and the ELASTIC_PASSWORD_FILE environment variable.
+     */
+    public void test81ConfigurePasswordThroughEnvironmentVariableFile() throws Exception {
+        // Test relies on configuring security
+        assumeTrue(distribution.isDefault());
+
+        final String xpackPassword = "hunter2";
+        final Path secretsDir = getTempDir().resolve("secrets");
+        final String passwordFilename = "password.txt";
+
+        try {
+            mkdir(secretsDir);
+
+            // ELASTIC_PASSWORD_FILE
+            Files.writeString(secretsDir.resolve(passwordFilename), xpackPassword + "\n");
+
+            Map<String, String> envVars = Map
+                .of(
+                    "ELASTIC_PASSWORD_FILE",
+                    "/run/secrets/" + passwordFilename,
+                    // Enable security so that we can test that the password has been used
+                    "xpack.security.enabled",
+                    "true"
+                );
+
+            // File permissions need to be secured in order for the ES wrapper to accept
+            // them for populating env var values
+            Files.setPosixFilePermissions(secretsDir.resolve(passwordFilename), p600);
+
+            final Map<Path, Path> volumes = Map.of(secretsDir, Path.of("/run/secrets"));
+
+            // Restart the container
+            runContainer(distribution(), volumes, envVars);
+
+            // If we configured security correctly, then this call will only work if we specify the correct credentials.
+            waitForElasticsearch("green", null, installation, "elastic", "hunter2");
```

Comment:
We might want to wrap the exception here to add some additional context.
Imagine this fails 6 months from now, will it be obvious from the failure that the passwrod was not applied or will it look as if elasticsearch didn't start ? I'm not sure what the error looks like, guess it's better if it's an authentication failure, but the additional context certainly doesn't hurt.